Next issue:
    $ /dspace/bin/dspace oai import -c                                                                                                                                                                                
    OAI 2.0 manager action started
    org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.dspace.xoai.app.XOAI': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.dspace.authorize.service.AuthorizeService org.dspace.xoai.app.XOAI.authorizeService; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type [org.dspace.authorize.service.AuthorizeService] found for dependency: expected at least 1 bean which qualifies as autowire candidate for this dependency. Dependency annotations: {@javax.inject.Inject()}
            at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:298)
            at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1148)
            at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireBean(AbstractAutowireCapableBeanFactory.java:299)
            at org.dspace.xoai.app.XOAI.main(XOAI.java:382)
            at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
            at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
            at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
            at java.lang.reflect.Method.invoke(Method.java:606)
            at org.dspace.app.launcher.ScriptLauncher.runOneCommand(ScriptLauncher.java:226)
            at org.dspace.app.launcher.ScriptLauncher.main(ScriptLauncher.java:78)
    Caused by: org.springframework.beans.factory.BeanCreationException: Could not autowire field: private org.dspace.authorize.service.AuthorizeService org.dspace.xoai.app.XOAI.authorizeService; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type [org.dspace.authorize.service.AuthorizeService] found for dependency: expected at least 1 bean which qualifies as autowire candidate for this dependency. Dependency annotations: {@javax.inject.Inject()}
            at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:531)
            at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88)
            at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:295)
            ... 9 more
    Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type [org.dspace.authorize.service.AuthorizeService] found for dependency: expected at least 1 bean which qualifies as autowire candidate for this dependency. Dependency annotations: {@javax.inject.Inject()}
            at org.springframework.beans.factory.support.DefaultListableBeanFactory.raiseNoSuchBeanDefinitionException(DefaultListableBeanFactory.java:997)
            at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:867)
            at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:779)
            at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:503)
            ... 11 more

--- a/dspace-api/src/main/java/org/dspace/content/dao/impl/ItemDAOImpl.java
+++ b/dspace-api/src/main/java/org/dspace/content/dao/impl/ItemDAOImpl.java
@@ -49,14 +49,24 @@ public class ItemDAOImpl extends AbstractHibernateDSODAO<Item> implements ItemDA
             boolean withdrawn, boolean discoverable, Date lastModified)
             throws SQLException
     {
-        Query query = createQuery(context, "SELECT i FROM Item i"
-                + " WHERE (inArchive = :in_archive OR withdrawn = :withdrawn)"
-                + "  AND discoverable = :discoverable"
-                + "  AND last_modified > :last_modified");
+        StringBuilder queryStr = new StringBuilder();
+        queryStr.append("SELECT i FROM Item i");
+        queryStr.append(" WHERE (inArchive = :in_archive OR withdrawn = :withdrawn)");
+        queryStr.append(" AND discoverable = :discoverable");
+
+        if(lastModified != null)
+        {
+            queryStr.append(" AND last_modified > :last_modified");
+        }
+
+        Query query = createQuery(context, queryStr.toString());
         query.setParameter("in_archive", archived);
         query.setParameter("withdrawn", withdrawn);
         query.setParameter("discoverable", discoverable);
-        query.setDate("last_modified", lastModified);
+        if(lastModified != null)
+        {
+            query.setTimestamp("last_modified", lastModified);
+	}
         return iterate(query);
     }
 
@@ -151,7 +161,7 @@ public class ItemDAOImpl extends AbstractHibernateDSODAO<Item> implements ItemDA
             throws SQLException
     {
         Query query = createQuery(context, "SELECT i FROM item i WHERE last_modified > :last_modified");
-        query.setDate("last_modified", since);
+        query.setTimestamp("last_modified", since);
         return iterate(query);
     }
 }

--- a/dspace-api/src/main/java/org/dspace/content/dao/impl/ItemDAOImpl.java
+++ b/dspace-api/src/main/java/org/dspace/content/dao/impl/ItemDAOImpl.java
@@ -56,7 +56,7 @@ public class ItemDAOImpl extends AbstractHibernateDSODAO<Item> implements ItemDA
         query.setParameter("in_archive", archived);
         query.setParameter("withdrawn", withdrawn);
         query.setParameter("discoverable", discoverable);
-        query.setParameter("last_modified", lastModified);
+        query.setDate("last_modified", lastModified);
         return iterate(query);
     }
 
@@ -151,7 +151,7 @@ public class ItemDAOImpl extends AbstractHibernateDSODAO<Item> implements ItemDA
             throws SQLException
     {
         Query query = createQuery(context, "SELECT i FROM item i WHERE last_modified > :last_modified");
-        query.setParameter("last_modified", since);
+        query.setDate("last_modified", since);
         return iterate(query);
     }
 }

--- a/dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java
+++ b/dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java
@@ -11,6 +11,19 @@ import com.lyncode.xoai.dataprovider.exceptions.ConfigurationException;
 import com.lyncode.xoai.dataprovider.exceptions.MetadataBindException;
 import com.lyncode.xoai.dataprovider.exceptions.WritingXmlException;
 import com.lyncode.xoai.dataprovider.xml.XmlOutputContext;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.ConnectException;
+import java.sql.SQLException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import javax.xml.stream.XMLStreamException;
+
 import org.apache.commons.cli.CommandLine;
 import org.apache.commons.cli.CommandLineParser;
 import org.apache.commons.cli.Options;
@@ -23,8 +36,18 @@ import org.apache.solr.client.solrj.SolrServer;
 import org.apache.solr.client.solrj.SolrServerException;
 import org.apache.solr.common.SolrDocumentList;
 import org.apache.solr.common.SolrInputDocument;
+
+import org.dspace.authorize.factory.AuthorizeServiceFactory;
 import org.dspace.authorize.service.AuthorizeService;
-import org.dspace.content.*;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Bundle;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.MetadataField;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.ItemService;
 import org.dspace.core.ConfigurationManager;
 import org.dspace.core.Constants;
 import org.dspace.core.Context;
@@ -33,29 +56,14 @@ import org.dspace.xoai.services.api.cache.XOAICacheService;
 import org.dspace.xoai.services.api.cache.XOAIItemCacheService;
 import org.dspace.xoai.services.api.cache.XOAILastCompilationCacheService;
 import org.dspace.xoai.services.api.config.ConfigurationService;
-import org.dspace.xoai.services.api.config.XOAIManagerResolver;
-import org.dspace.xoai.services.api.context.ContextService;
 import org.dspace.xoai.services.api.CollectionsService;
 import org.dspace.xoai.services.api.solr.SolrServerResolver;
 import org.dspace.xoai.solr.DSpaceSolrSearch;
 import org.dspace.xoai.solr.exceptions.DSpaceSolrException;
 import org.dspace.xoai.solr.exceptions.DSpaceSolrIndexerException;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 
-import javax.xml.stream.XMLStreamException;
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.net.ConnectException;
-import java.sql.SQLException;
-import java.text.ParseException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Date;
-import java.util.Iterator;
-import java.util.List;
-import javax.inject.Inject;
-import org.dspace.content.service.ItemService;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 
 import static com.lyncode.xoai.dataprovider.core.Granularity.Second;
 import static org.dspace.xoai.util.ItemUtils.retrieveMetadata;
@@ -75,21 +83,14 @@ public class XOAI {
     @Autowired
     private SolrServerResolver solrServerResolver;
     @Autowired
-    private XOAIManagerResolver xoaiManagerResolver;
-    @Autowired
-    private ContextService contextService;
-    @Autowired
     private XOAILastCompilationCacheService xoaiLastCompilationCacheService;
-    @Autowired
-    private XOAICacheService xoaiCacheService;
     @Autowired
     private XOAIItemCacheService xoaiItemCacheService;
     @Autowired
     private CollectionsService collectionsService;
-    @Inject
-    private AuthorizeService authorizeService;
-    @Inject
-    private ItemService itemService;
+
+    private final AuthorizeService authorizeService;
+    private final ItemService itemService;
 
 
     private List<String> getFileFormats(Item item) {
@@ -113,11 +114,19 @@ public class XOAI {
         this.optimize = optimize;
         this.clean = clean;
         this.verbose = verbose;
+
+        // Load necessary DSpace services
+        this.authorizeService = AuthorizeServiceFactory.getInstance().getAuthorizeService();
+        this.itemService = ContentServiceFactory.getInstance().getItemService();
     }
 
     public XOAI(Context ctx, boolean hasOption) {
-        context = ctx;
-        verbose = hasOption;
+        this.context = ctx;
+        this.verbose = hasOption;
+
+        // Load necessary DSpace services
+        this.authorizeService = AuthorizeServiceFactory.getInstance().getAuthorizeService();
+        this.itemService = ContentServiceFactory.getInstance().getItemService();
     }
 
     private void println(String line) {

